examples/tests/cloud-push-pull/Earthfile

```diff
@@ -0,0 +1,136 @@
+    
+    ENV PATH $PATH:/test/google-cloud-sdk/bin
+
+    RUN --secret=GCP_KEY=+secrets/earthly-technologies/gcp/ci-cd-key \
+        echo $GCP_KEY > key.json
+
+    RUN gcloud auth activate-service-account --key-file /test/key.json
+
+google-artifact-registry:
+    FROM +google-base
+    COPY google-artifact-repository.earth ./Earthfile
+
+    RUN gcloud auth configure-docker us-west1-docker.pkg.dev
+
+    RUN --privileged \
+    --entrypoint \
+    --mount=type=tmpfs,target=/tmp/earthly \
+    -- --ci --push +push
+
+    RUN --privileged \
+    --entrypoint \
+    --mount=type=tmpfs,target=/tmp/earthly \
+    -- -P +pull
+
+google-container-repository:
+    FROM +google-base
+    COPY google-container-registry.earth ./Earthfile
+
+    RUN gcloud auth configure-docker
+
+    RUN --privileged \
+    --entrypoint \
+    --mount=type=tmpfs,target=/tmp/earthly \
+    -- --ci --push +push
+
+    RUN --privileged \
+    --entrypoint \
+    --mount=type=tmpfs,target=/tmp/earthly \
+    -- -P +pull
+
+azure-container-registry:
+    COPY azure-container-registry.earth ./Earthfile
+   
+   # Note that we dont have to install a cred helper here, also the cred helper wouldnt help here anyways
+    RUN --secret AZ_USERNAME=+secrets/earthly-technologies/azure/ci-cd-username \
+        --secret AZ_PASSWORD=+secrets/earthly-technologies/azure/ci-cd-password \
+        docker login earthlyintegrationtest.azurecr.io --username $AZ_USERNAME --password $AZ_PASSWORD
+
+    RUN --privileged \
+    --entrypoint \
+    --mount=type=tmpfs,target=/tmp/earthly \
+    -- --ci --push +push
+
+    RUN --privileged \
+    --entrypoint \
+    --mount=type=tmpfs,target=/tmp/earthly \
+    -- -P +pull
+
+amazon-base:
+    ENV GLIBC_VER=2.31-r0
+
+    # install glibc compatibility for alpine
+    RUN apk --no-cache add \
+        binutils \
+        curl \
+    && curl -sL https://alpine-pkgs.sgerrand.com/sgerrand.rsa.pub -o /etc/apk/keys/sgerrand.rsa.pub \
+    && curl -sLO https://github.com/sgerrand/alpine-pkg-glibc/releases/download/${GLIBC_VER}/glibc-${GLIBC_VER}.apk \
+    && curl -sLO https://github.com/sgerrand/alpine-pkg-glibc/releases/download/${GLIBC_VER}/glibc-bin-${GLIBC_VER}.apk \
+    && apk add --no-cache \
+        glibc-${GLIBC_VER}.apk \
+        glibc-bin-${GLIBC_VER}.apk \
+    && curl -sL https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip -o awscliv2.zip \
+    && unzip awscliv2.zip \
+    && aws/install \
+    && rm -rf \
+        awscliv2.zip \
+        aws \
+        /usr/local/aws-cli/v2/*/dist/aws_completer \
+        /usr/local/aws-cli/v2/*/dist/awscli/data/ac.index \
+        /usr/local/aws-cli/v2/*/dist/awscli/examples \
+    && apk --no-cache del \
+        binutils \
+        curl \
+    && rm glibc-${GLIBC_VER}.apk \
+    && rm glibc-bin-${GLIBC_VER}.apk \
+    && rm -rf /var/cache/apk/*
+
+    RUN apk add go
+    RUN go get -u github.com/awslabs/amazon-ecr-credential-helper/ecr-login/cli/docker-credential-ecr-login
```

Comment:
Also have to install from source since thats really the only way from their installation instructions that works for us here.